Client/Frontend/Browser/Certificate Viewer/CertificateViewController.swift

```diff
@@ -0,0 +1,527 @@
+        throw OIDConversionError.invalidBEREncoding
+      }
+      
+      // Drop first 2 octets as it isn't needed for the calculation
+      var X = UInt32(oid[2]) / 40
+      let Y = UInt32(oid[2]) % 40
+      var sub = UInt64(0)
+      
+      var dot_notation = String()
+      if X > 2 {
+        X = 2
+        
+        dot_notation = "\(X)"
+        if (UInt32(oid[2]) & 0x80) != 0x00 {
+          sub = 80
+        } else {
+          dot_notation = ".\(Y + ((X - 2) * 40))"
+        }
+      } else {
+        dot_notation = "\(X).\(Y)"
+      }
+      
+      // Drop first 2 octets as it isn't needed for the calculation
+      // Start at the next octet
+      var value = UInt64(0)
+      for i in (sub != 0 ? 2 : 3)..<oid.count {
+        value = (value << 7) | (UInt64(oid[i]) & 0x7F)
+        if (UInt64(oid[i]) & 0x80) != 0x80 {
+          dot_notation += ".\(value - sub)"
+          sub = 0
+          value = 0
+        }
+      }
+      return dot_notation
+    }
+    
+    static func absolute_oid_to_oid(oid: String) -> Data {
+        do {
+            let absolute_oid: [UInt8] = try absolute_oid_to_oid(oid: oid)
+            return Data(bytes: absolute_oid, count: absolute_oid.count)
+        } catch {
+            return Data()
+        }
+    }
+
+    static func oid_to_absolute_oid(oid: Data) -> String {
+        do {
+            return try oid_to_absolute_oid(oid: oid.getBytes())
+        } catch {
+            return String()
+        }
+    }
+}
+
+struct CertificateTitleView: View {
+    let isRootCertificate: Bool
+    let commonName: String
+    
+    var body: some View {
+        HStack(alignment: .center, spacing: 15.0) {
+            Image(uiImage: isRootCertificate ? #imageLiteral(resourceName: "Root") : #imageLiteral(resourceName: "Other"))
+            VStack(alignment: .leading, spacing: 10.0) {
+                Text(commonName)
+                    .font(.system(size: 16.0, weight: .bold))
+            }
+        }.background(Color(UIColor.secondaryBraveGroupedBackground))
+    }
+}
+
+struct CertificateKeyValueView: View, Hashable {
+    let title: String
+    let value: String?
+    
+    init(title: String, value: String? = nil) {
+        self.title = title
+        self.value = value
+    }
+    
+    var body: some View {
+        HStack(alignment: .center, spacing: 12.0) {
+            Text(title)
+                .font(.system(size: 12.0))
+            Spacer()
+            if let value = value, !value.isEmpty {
+                Text(value)
+                    .lineLimit(nil)
+                    .fixedSize(horizontal: false, vertical: true)
+                    .font(.system(size: 12.0, weight: .medium))
+            }
+        }
+    }
+}
+
+struct CertificateSectionView<ContentView>: View where ContentView: View {
+    let title: String
+    let values: [ContentView]
+    
+    var body: some View {
+        Section(header: Text(title)
+                    .font(.system(size: 12.0))) {
```

Comment:
Same nits as above: center is default, not hardcoded font sizes, lineLimit(nil) is default